File: mono/help.cs
// qucik swap data demo
// help.cs
using System;

public class Demo {
	public static void Main() {
		int x = 3;
		int y = 5;

		Swap(ref x, ref y);
		Console.WriteLine("x: {0} - y: {1}", x, y);
	}

	static void Swap(ref int a, ref int b) {
		int c = a;
		a = b;
		b =c;
		Console.WriteLine("a: {0} - b: {1}", a, b);
	}
}

File: mono/c/src/help.c
#include <stdio.h>

void swap(int *a, int *b)
{
	int c = *a;
	*a = *b;
	*b = c;
	printf("a: %d - b: %d\n", *a, *b);
}

int main()
{
	int x = 3;
	int y = 5;

	swap(&x, &y);
	printf("x: %d - y: %d\n", x, y);
	
	return 0;
}
